guides/runbooks/orca-quality-of-service.md

```diff
@@ -0,0 +1,107 @@
+When an execution is `BUFFERED`, it will effectively stay in a waiting state until it is unbuffered, which we'll go over later.
+
+`BufferPolicy` functions are pluggable and can contain arbitrary logic. 
+For example, one `BufferPolicy` that ships always enabled is [EnqueueDeckOrchestrationBufferPolicy][deck-buffer-policy], which will always `ENQUEUE` an execution it is an Orchestration and from the UI. 
+This specific policy forces the `ENQUEUE` status, even if other policies call for the execution to be buffered; this is done through a `force` flag that policies can return.
+An example of other pluggable behavior is determining buffering action based on criticality of the execution: At Netflix we have a custom concept of application criticality, so we can buffer low criticality executions to allow capacity for higher criticality executions.
+
+Once an execution is put into a `BUFFERED` state, it will remain in that state until the [ExecutionPromoter][promoter] decides to change it to `NOT_STARTED` and enqueue it for processing.
+Similar to the `ExecutionBufferActuator`, the `ExecutionPromoter` uses an ordered chain of [PromotionPolicy][promotion-policy] functions to determine what buffered executions to promote.
+Every promotion cycle, the list of buffered executions (called candidates) are passed through the policies, each reducing the list of candidates to a final list of executions that will be promoted.
+Again, these policies can have arbitrary logic, but by default a naive promotion policy is included that will promote the _N_ oldest executions.
+This promotion process happens (by default) on a 5-second interval on every Orca instance.
+
+With both `BufferPolicy` and `PromotionPolicy`, the results of each function returns a result with a human readable "reason", which is logged out for each execution that is evaluated so it is easy to trace.
+
+**Note**: This is the first implementation of the QoS system, we plan to iterate on this concept and make it more advanced over time.
+You can read the [original proposal][proposal to get an idea of a potential roadmap.
+
+# Configuration
+
+These configurations are not guaranteed to be fully inclusive of all knobs.
+A definitive list is available via the codebase.
+
+* `qos.enabled`: Boolean (default `false`). Global flag controlling whether or not the system is enabled. This flag will not disable the `ExecutionPromoter`.
+* `qos.learningMode.enabled`: Boolean (default `true`). If enabled, executions will always be `ENQUEUED`, but log messages & metrics will be emitted saying what the system would have done. This flag has no effect on `ExecutionPromoter`.
+* `pollers.qos.promoteIntervalMs`: Integer (default `5000`). The time (in milliseconds) that the promotion process will be run.
+
+## BufferPolicy: Naive
+
+The `NaiveBufferPolicy` will always buffer executions when enabled.
```

Comment:
is there just one bufferpolicy? or are we expecting to add more?

Reply:
Right now there's just one. We'll be adding our own in `-nlfx` for `AppCriticalityBufferPolicy`.